src/Microsoft.Health.Dicom.SqlServer/Features/Schema/Migrations/2.sql

```diff
@@ -364,6 +364,274 @@ CREATE NONCLUSTERED INDEX IX_ChangeFeed_StudyInstanceUid_SeriesInstanceUid_SopIn
     SopInstanceUid
 )
 
+/*************************************************************
+    Custom Tag Table
+    Stores added custom tags
+    TagPath is represented without any delimiters and each level takes 8 bytes
+    TagLevel can be 0, 1 or 2 to represent Instance, Series or Study level
+**************************************************************/
+CREATE TABLE dbo.CustomTag (
+    TagKey                  BIGINT               NOT NULL, --PK
+    TagPath                 VARCHAR(64)          NOT NULL,
+    TagVR                   VARCHAR(2)           NOT NULL,
+    TagLevel                TINYINT              NOT NULL
+)
+
+CREATE UNIQUE CLUSTERED INDEX IXC_CustomTag ON dbo.CustomTag
+(
+    TagKey
+)
+
+CREATE NONCLUSTERED INDEX IX_CustomTag_TagPath ON dbo.CustomTag
+(
+    TagPath
+)
+
+/*************************************************************
+    Custom Tag Data Table for VR Types mapping to String
+**************************************************************/
+CREATE TABLE dbo.CustomTagString (
+    TagKey                  BIGINT               NOT NULL, --PK
+    TagValue                VARCHAR(64)          NOT NULL,
+    StudyKey                BIGINT               NOT NULL, --FK
+    SeriesKey               BIGINT               NULL,     --FK
+    InstanceKey             BIGINT               NULL,     --FK
+    Watermark               BIGINT               NOT NULL
+) WITH (DATA_COMPRESSION = PAGE)
+
+CREATE UNIQUE CLUSTERED INDEX IXC_CustomTagString ON dbo.CustomTagString
+(
+    TagKey,
+    StudyKey,
+    SeriesKey,
+    InstanceKey
+)
+INCLUDE
+(
+    WaterMark
+)
+
+CREATE NONCLUSTERED INDEX IXC_CustomTagString_TagKey_StudyKey ON dbo.CustomTagString
+(
+    TagKey,
+    StudyKey,
+)
+
+CREATE NONCLUSTERED INDEX IXC_CustomTagString_TagKey_SeriesKey ON dbo.CustomTagString
+(
+    TagKey,
+    SeriesKey,
+)
+
+CREATE NONCLUSTERED INDEX IXC_CustomTagString_TagKey_InstanceKey ON dbo.CustomTagString
+(
+    TagKey,
+    InstanceKey,
+)
+
+CREATE NONCLUSTERED INDEX IXC_CustomTagString_TagKey_TagValue ON dbo.CustomTagString
+(
+    TagKey,
+    TagValue
+)
+
+/*************************************************************
+    Custom Tag Data Table for VR Types mapping to Int
+**************************************************************/
+CREATE TABLE dbo.CustomTagInt (
+    TagKey                  BIGINT               NOT NULL, --PK
+    TagValue                BIGINT               NOT NULL,
+    StudyKey                BIGINT               NOT NULL, --FK
+    SeriesKey               BIGINT               NULL,     --FK
+    InstanceKey             BIGINT               NULL,     --FK
```

Comment:
>FK [](start = 61, length = 2)

Do we avoid declaring foreign key constraints to avoid performance penalties when inserting?